### PR TITLE
Move activation into conda-env

### PR DESCRIPTION
**Pull request by tswicegood**
This provides a backwards compatible method of moving activation over to [conda-env][].  It requires conda-env v1.2.0 to be installed, but once that's installed everything works as it did + the new activation behavior.

[conda-env]: https://github.com/conda/conda-env

### CODE REVIEW

**Comment by tswicegood** [comment]
I just released conda-env 1.4.1 which fixes a bug around the way conda's `activate.bat` script detects (see https://github.com/conda/conda-env/commit/d8a6659f20bf0a3686fe4a373e609aed853ebef8).  That can easily be ported to conda, but there are two problems:

* I don't have a way to release a bug fix to conda
* If this PR is merged and released, the bug fix is already out there in the wild

As I understood conda-env the goal was to move environment management to that package (at least until such time as it can / should be merged back into conda.  If that's the case, we should merge this and start guiding users who use this functionality over toward the new package (which is what this does).

Added benefit, we'll get a pretty good idea how many people use the activation scripts as we can watch conda-env download numbers from binstar. :-)

cc @teoliphant @pzwang

**Comment by tswicegood** [comment]
@asmeurer / @ilanschnell: another option we have here is to add a dependency on `conda-env` to conda so that an upgrade automatically installed conda-env as well.  That solves the problem of users' workflows being interrupted as `activate` and `deactivate` will continue to work, but with the new features of conda-env added in.

File: bin/activate [hunk 1]
```diff
@@ -1,67 +1,7 @@
...
-    fi
-else
-    return $?
-fi
-
-if [[ -n $BASH_VERSION ]]; then
-    hash -r
-elif [[ -n $ZSH_VERSION ]]; then
-    rehash
+ROOT_PREFIX=$(conda info --json  | grep root_prefix | awk -F ':' '{print $2}' | awk -F '"' '{print $2}')
```
Comment (asmeurer) [inline bin/activate]: I think you can use the placeholder prefix here.

File: bin/activate [hunk 2]
```diff
@@ -1,67 +1,7 @@
...
+if [[ -a "${ROOT_PREFIX}/bin/env-activate" ]]; then
+    source "${ROOT_PREFIX}/bin/env-activate" $@
 else
-    echo "Only bash and zsh are supported"
-    return 1
+    source "${ROOT_PREFIX}/bin/_conda-env-missing"
```
Comment (asmeurer) [inline bin/activate]: Instead of doing this, just put the original activate (that is there right now).

**Comment by teoliphant** [comment]
@tswicegood is going to fix the language when you don't have conda-env to be more empathetic and then we can merge this.  We will also need to add conda-env to repo.continuum.io so that *anyone* who can update conda with this change will also be able to get conda-env. 

We will need to work over the coming weeks to merge conda-env into conda as well and then this "wart" goes away entirely.

**Comment by tswicegood** [comment]
As of 0d81be5ebf96ec37f59aace8faca8bad3874ea0a, this message reads:

> We're making conda environments even better!  Part of this process
> is changing the way environment activation works.  You need to
> install the new conda-env package before you can use these feature:
>
>     conda install -c conda conda-env
>
> For more information, please visit: https://github.com/conda/conda-env

File: bin/activate (hunk 1 above)
Comment (tswicegood) [inline bin/activate]: I'm not sure what you mean by "placeholder prefix" here.  This is setting the root conda environment which is hate only place the `_conda-env-missing` file is guaranteed to be located.

**Comment by asmeurer** [comment]
I think the "changing the way environment activation works" part is confusing. I've never been clear on what that has meant whenever you have said it.

File: bin/activate (hunk 1 above)
Comment (asmeurer) [inline bin/activate]: If you use `/opt/anaconda1anaconda2anaconda3` then conda will replace it with the root prefix when it is installed. Alternately you can inject the build environment here when installing. 

This works too, it's just the placeholder way requires less mundging.

**Pull request merged**